File: backend/app/schemas/medication.py
from datetime import datetime, time
from typing import Optional

from pydantic import BaseModel


# ==========================================================
# Create Medication
# ==========================================================

class MedicationCreate(BaseModel):
    medicine_name: str
    dosage: Optional[str] = None
    reminder_time: Optional[time] = None

    before_food: bool = False

    morning: bool = False
    afternoon: bool = False
    evening: bool = False
    night: bool = False


# ==========================================================
# Medication Response
# ==========================================================

class MedicationResponse(BaseModel):
    id: int
    patient_id: int
    medicine_name: str
    dosage: Optional[str] = None
    reminder_time: Optional[time] = None

    before_food: bool

    morning: bool
    afternoon: bool
    evening: bool
    night: bool

    active: bool

    class Config:
        from_attributes = True


# ==========================================================
# Medication Status Response
# Used by Family + Caregiver Dashboards
# ==========================================================

class MedicationStatusResponse(BaseModel):
    id: int
    patient_id: int
    medicine_name: str
    dosage: Optional[str] = None
    reminder_time: Optional[time] = None

    before_food: bool

    morning: bool
    afternoon: bool
    evening: bool
    night: bool

    active: bool

    # Today's medication status
    status: str

    # Medication log information
    given_by: Optional[str] = None
    given_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: frontend/services/api/medication.ts
import API from "./axios";

export interface Medication {
  id: number;
  patient_id: number;
  medicine_name: string;
  dosage?: string | null;
  reminder_time?: string | null;
  before_food: boolean;
  morning: boolean;
  afternoon: boolean;
  evening: boolean;
  night: boolean;
  active: boolean;

  status: "taken" | "pending" | "upcoming";
  given_by?: string | null;
  given_at?: string | null;
}

// ==========================================================
// FAMILY
// ==========================================================

export async function getMyMedications(): Promise<Medication[]> {
  const response = await API.get("/medications/me");

  return response.data;
}

// ==========================================================
// CAREGIVER
// ==========================================================

export async function getCaregiverMedications(): Promise<
  Medication[]
> {
  const response = await API.get("/medications/caregiver");

  return response.data;
}

// ==========================================================
// CAREGIVER
// MARK MEDICATION AS GIVEN
// ==========================================================

export async function markMedicationAsGiven(
  medicationId: number
) {
  const response = await API.post(
    `/medications/${medicationId}/given`
  );

  return response.data;
}